src/lib/components/Table/Table.js

```diff
@@ -0,0 +1,199 @@
+    }
+  }
+
+  getNewSortCondition(columnKey) {
+    const { sortCondition } = this.state;
+    const newSortCondition = {};
+    const options = ['ascending', 'descending', 'none'];
+
+    // Move to next option in options array
+    newSortCondition[columnKey] = (
+      options[(options.indexOf(sortCondition[columnKey]) + 1) % options.length]
+    );
+
+    this.setState({ sortCondition: newSortCondition }, () => {
+      this.sortData(columnKey);
+    });
+  }
+
+  sortData(columnKey) {
+    const { data, sortCondition } = this.state;
+
+    if (sortCondition[columnKey] === 'none') {
+      const unsortedData = this.state.originalData;
+      this.setState({ data: [...unsortedData] });
+    } else {
+      const sortedData = data.sort((a, b) => {
+        const x = a[columnKey];
+        const y = b[columnKey];
+        return (
+          sortCondition[columnKey] === 'ascending' ? naturalCompare(x, y) : naturalCompare(y, x)
+        );
+      });
+
+      this.setState({ data: sortedData });
+    }
+  }
+
+  generateData() {
+    const { children } = this.props;
+    const columns = [];
+    const data = [];
+
+    children.forEach((row, rowIndex) => {
+      if (row.props.tableHeader) {
+        row.props.children.forEach((cell, cellIndex) => {
+          let value = '';
+          if (typeof cell.props.children === 'string') value = cell.props.children;
+          columns.push({
+            key: `column${cellIndex}`,
+            label: value,
+            align: cell.props.align,
+            header: cell.props.columnHeader,
+          });
+        });
+      } else {
+        const index = rowIndex - 1;
+        data[index] = {};
+        data[index].id = rowIndex;
+        row.props.children.forEach((cell, cellIndex) => {
+          let value = '';
+          if (typeof cell.props.children === 'string') value = cell.props.children;
+          data[index][`column${cellIndex}`] = value;
+        });
+      }
+    });
+
+    this.setState({
+      columns,
+      data: [...data],
+      originalData: data,
+    });
+  }
+
+  generateHeader() {
+    const { columns, sortCondition } = this.state;
+    const { sortable } = this.props;
+
+    const columnHeaders = columns.map((headerData) => {
+      const ariaSort = sortCondition[headerData.key];
+      const onClick = sortable ? () => this.getNewSortCondition(headerData.key) : null;
+      return (
+        <TableCell
+          columnHeader
+          key={headerData.key}
+          ariaSort={ariaSort}
+          align={headerData.align}
+          onClick={onClick}
+        >
+          {headerData.label}
+        </TableCell>);
+    });
+
+    const header = (
+      <thead>
+        <TableRow key="header-row">
+          {columnHeaders}
+        </TableRow>
+      </thead>
+    );
+
+    return header;
+  }
+
+  generateRows() {
+    const { columns, data } = this.state;
+    const rows = data.map((item) => {
+      const cells = columns.map(headerData => (
```

Comment:
I'd personally break this map out to a 'generateCells()' or something, I don't like immediately nested loopy boys.
`const cells = this.generateCells(columns);`